### ISSUE TITLE
Cost Management + OpenShift is not using token authentication type, some authentication types does not create authentication entities

### ISSUE THREAD

**Issue by rvsia** [RedHatInsights/sources-api#214]
**A.** According to `application_types` seed, cost management is using token for OpenShift

https://github.com/RedHatInsights/sources-api/blob/master/db/seeds/application_types.yml#L24

However, it is using only `Cluster ID` a.k.a. `source.ref`, not the `Token` a.k.a. `authentication.password`.

**B.** Also, because this authentication does not create `endpoint`/`authentication` entities, it's not possible to create the join `authentication_application` entity.

The same applies for Satellite `receptore_node` authentication type, that creates only an `endpoint` entity, so it's also not possible to create the join table for applications. (However, by a mistake it creates empty `authentication` entity with `authtype` set.)

Without the `authentication_application` entities, applications cannot be sure that the source has required values to authenticate.

**Comment by rvsia** [comment]
@gmcculloug @lindgrenj6  I think this issue can be related to the issue with no endpoints for cost management ocp sources, as there are additional issues that we should consider when no endpoint/auth entities are created.

**Comment by gmcculloug** [comment]
Thanks @rvsia, this help me understand the issue better.  It also makes me realize that we will likely have the same issue with the Catalog receptor as it is following the Satellite approach.  cc @mkanoor 

I think we might need to have a discussion about this next week when @syncrou is back as it is unclear to me what exactly we want to try to support.

All this leads back to an issue performing the availability check.  (See [BZ 1854107](https://bugzilla.redhat.com/show_bug.cgi?id=1854107))
If I understanding the current situation correctly, Cost only stores the Cluster ID in the Sources service, so we have no way to validate it.  It seems that we might want to honor this and rely on the application availability status.

However, Satellite, and soon Catalog, provides the Receptor ID so we have enough information to contact and validate that Source.

Pinging a few others in hopes they can help provide some clarity or guidance: @slemrmartin @gtanzillo @agrare

**Comment by lindgrenj6** [comment]
Just to extend on the receptor approach, we store the `receptor_node` in an endpoint object so our availability checks will work if there is an endpoint with the receptor node set and the operations worker has been modified to support receptor. We also don't need an authentication object since we just use that receptor node id and the external tenant number to communicate with the platform receptor controller. 

TL;DR is once we add the necessary code to support receptor for each operations worker it should work.
